integrationTests/multiShard/smartContract/scCallingSC_test.go

```diff
@@ -18,10 +19,112 @@ import (
 	"github.com/ElrondNetwork/elrond-go/process/factory"
 	"github.com/ElrondNetwork/elrond-go/sharding"
 	factory2 "github.com/ElrondNetwork/elrond-go/vm/factory"
-	"github.com/ElrondNetwork/elrond-vm-common"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
 	"github.com/stretchr/testify/assert"
 )
 
+func TestSCCallingInIntraShard(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	_ = logger.SetLogLevel("*:INFO")
+
+	numOfShards := 1
+	nodesPerShard := 4
+	numMetachainNodes := 0
+
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+
+	nodes := integrationTests.CreateNodes(
+		numOfShards,
+		nodesPerShard,
+		numMetachainNodes,
+		integrationTests.GetConnectableAddress(advertiser),
+	)
+
+	idxProposers := make([]int, numOfShards+1)
+	for i := 0; i < numOfShards; i++ {
+		idxProposers[i] = i * nodesPerShard
+	}
+	idxProposers[numOfShards] = numOfShards * nodesPerShard
+
+	integrationTests.DisplayAndStartNodes(nodes)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Node.Stop()
+		}
+	}()
+
+	initialVal := big.NewInt(10000000000000)
+	initialVal.Mul(initialVal, initialVal)
+	fmt.Printf("Initial minted sum: %s\n", initialVal.String())
+	integrationTests.MintAllNodes(nodes, initialVal)
+
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	// mint smart contract holders
+	firstSCOwner := []byte("12345678901234567890123456789000")
+	secondSCOwner := []byte("99945678901234567890123456789001")
+
+	mintPubKey(firstSCOwner, initialVal, nodes)
+	mintPubKey(secondSCOwner, initialVal, nodes)
+
+	// deploy the smart contracts
+	firstSCAddress := putDeploySCToDataPool("./testdata/first/first.wasm", firstSCOwner, 0, big.NewInt(50), nodes)
+	//000000000000000005005d3d53b5d0fcf07d222170978932166ee9f3972d3030
+	secondSCAddress := putDeploySCToDataPool("./testdata/second/second.wasm", secondSCOwner, 0, big.NewInt(50), nodes)
+	//00000000000000000500017cc09151c48b99e2a1522fb70a5118ad4cb26c3031
+
+	// Run two rounds, so the two SmartContracts get deployed.
+	integrationTests.UpdateRound(nodes, round)
```

Comment:
make one function out of lines 86 to 90. you have the function in systemVM tests. Make it public and put in initTestProcessorNode.go

Reply:
with that you can remove a lot of duplicated code.

Reply:
Moved `WaitForOperationToBeDone` into `testInitializer.go` and used it instead of looping over Rounds.